packet_injector/server.go

```diff
@@ -40,7 +40,23 @@ const (
 
 // PacketInjectorServer creates a packet injector server API
 type PacketInjectorServer struct {
-	Graph *graph.Graph
+	Graph    *graph.Graph
+	Channels *Channels
+}
+
+func (pis *PacketInjectorServer) stopPI(msg *shttp.WSJSONMessage) error {
+	var uuid string
+	if err := common.JSONDecode(bytes.NewBuffer([]byte(*msg.Obj)), &uuid); err != nil {
+	}
+	pis.Channels.Lock()
+	c, ok := pis.Channels.Pipes[uuid]
+	pis.Channels.Unlock()
+	if ok {
+		c <- true
+		return nil
+	} else {
```

Comment:
this else is useless

Reply:
this will be useful if the PI is not running in the agent(no channel to communicate to the goroutine) and the corresponding entry is present in the analyzer.